### ISSUE TITLE
Repeal personal exemption for dependents under age 18. 

### ISSUE THREAD

**Issue by MattHJensen** [open-source-economics/Tax-Calculator#1409]
Question from a Senate staffer:

> I was wondering if OSPC had anything in the works to change the personal exemption in greater detail, or if you had any tips for how to model a proposed reform in Tax Brain. Specifically, we’d like to see the effects of repealing personal exemptions for dependents under age 18. TPC has a score from last year here: http://www.taxpolicycenter.org/model-estimates/options-repeal-child-tax-credit-and-exemption-dependents-july-2016/t16-0134-options

I asked if they have a deadline:

> It would be ideal to have before the 20th. Should have asked sooner, as we’ve just been plugging in the TPC number for our assumptions on a project, but it’s really apples-to-oranges considering we’re using tax brain for everything else.

@hdoupe, could you take a look at this? 

cc @andersonfrailey

**Comment by hdoupe** [comment]
I think we can do that with the new python generated matched file.  There are variables for the number of dependents under 5, under 13, under 18, between 18 and 21, above 21, and above 65.  Then, we could pass those to the AGI function and do something like

`pre_c04600 = (XTOT - n18) * II_em`

But we would need the python generated match file for that.

@MattHJensen @andersonfrailey

**Comment by MattHJensen** [comment]
@hdoupe, could we add `nu18` to the current puf.csv with the same method that we added `nu05`? I hope that testing and adopting the new python-generated match file need not block this request.

**Comment by hdoupe** [comment]
@MattHJensen Yes, we can do that.  We have a variable for the age range of the first three dependents, sorted by age from youngest to oldest.  We can get the number of dependents under 18 from that, but we may miss some if someone has more than three dependents.

**Comment by andersonfrailey** [comment]
@MattHJensen I'm also running the python scripts right now to test out the resulting PUF file so we should the python generated file soon.

**Comment by hdoupe** [comment]
I'm looking at the age breakdown now.  It looks like we can do either "13 under 17" or "17 under 19."  To me, that means 13-16, inclusive for the first group and 17, 18 for the second group.  If we go with "13 under 17" then 17 year olds will be excluded but if we go "17 under 19" then 18 year olds will be included.  

My interpretation of the request is that it wants to account for dependents strictly less than 18 years of age.  I can't think of a substantial reason to pick one category over the other for the cutoff.  What do you all think?

@MattHJensen @andersonfrailey

**Comment by MattHJensen** [comment]
@hdoupe, these discussions might be helpful:

https://github.com/open-source-economics/taxdata/pull/41
https://github.com/open-source-economics/taxdata/pull/43

**Comment by hdoupe** [comment]
@MattHJensen Thanks, those helped a lot.  

I used the CPS to find the proportion of 17 and 18 year old dependents who are 18 years old.  Then I removed members from the puf's "17 under 19" age group according to that proportion.  

I also looked at the proportion of tax units that have at least three dependents under the age of 18 to see how many had 4 or more dependents under 18.  I then tried to model that by adding either 0, 1, or 2 under age 18 dependents to the current count according to the distribution observed in the CPS.  However, after looking into the puf variable definitions further,  I saw that the range for the number of dependents is set to 0 - 3.  To account for this, I would have to adjust XOCAH,  XOCAWH, and XTOT.  So, I'm thinking that adding on more dependents could add an unnecessary amount of complexity.  

I'll open a PR in taxdata and reference this issue there.

**Comment by martinholmer** [comment]
@MattHJensen said a few days ago in issue #1409:
> Question from a Senate staffer:
>
> I was wondering if OSPC had anything in the works to change the personal exemption in greater detail, or if you had any tips for how to model a proposed reform in Tax Brain.  Specifically, we’d like to see the effects of repealing personal exemptions for dependents under age 18.

So, does new Tax-Calculator logic need to be added in order to simulate this kind of reform?

@hdoupe

**Comment by hdoupe** [comment]
@martinholmer I think that we just have to subtract `nu18` from `XTOT` when the personal exemption is calculated in the AGI function in `functions.py`.

`XTOT` is used to calculate the child tax credit and lump sum tax.  But, it looks like this request is only concerned with the personal exemption tax. 

So, I would add a policy parameter like : `_II_nu18`.  This would be a boolean variable.  Then in the AGI function, do something like:
`if II_nu18 == True: XTOT = XTOT - nu18`

**Comment by martinholmer** [comment]
@hdoupe described code needed to implement reform requested in issue #1409.

Yes, I think your basic idea is sound.
But I'd suggest a longer policy parameter name. 
Maybe, something like `_II_no_em_nu18`. 
What do you think?

Here is the way the code is now:
```
    # calculate personal exemption amount
    pre_c04600 = XTOT * II_em
    if DSI:
        pre_c04600 = 0.
    # phase-out personal exemption amount
```
So, it could be changed to be something like:
```
    # calculate personal exemption amount
    if II_no_em_nu18:
        pre_c04600 = max(0, XTOT - nu18) * II_em
    else:
        pre_c04600 = XTOT * II_em
    if DSI:
        pre_c04600 = 0.
    # phase-out personal exemption amount
```
Not sure how paranoid to be about the input data.  Maybe we don't need to max(0,...) business.
What do you think?

Why don't you prepare a Tax-Calculator pull request along these lines?

@MattHJensen @feenberg

**Comment by hdoupe** [comment]
@martinholmer Looks good to me.  I'll keep the max function to be on the safe side.

**Comment by martinholmer** [comment]
Merging pull request #1428 by @hdoupe has resolved issue #1409, which is now being closed.
